### PR TITLE
Add config option to gate pam_authenticate calls

### PR DESCRIPTION
**Pull request by awly**
Most users won't need this, so the behavior is optional. Default system
configs will usually trigger a password prompt, which is why this
feature is disabled by default.

Updates https://github.com/gravitational/teleport/issues/4551

### CODE REVIEW

File: docs/4.4/features/ssh-pam.md [hunk 1]
```diff
@@ -286,3 +289,17 @@ ssh_service:
...
 Now attempting to login as an existing user should result in the creation of the
 user and a successful login.
+
+## Additional authentication steps
+
+Using the PAM `auth` modules, it is possible to add additional authentication
+steps during user login. These can include passwords, 2nd factor or even
+biometrics.
+
+Note that Teleport enables strong SSH authentication out of the box using
```
Comment (benarent) [inline docs/4.4/features/ssh-pam.md:299]: 👍 - nice note.

**Review by benarent — APPROVED** [review]

File: docs/4.4/features/ssh-pam.md [hunk 2]
```diff
@@ -57,6 +57,9 @@ teleport:
          enabled: true
          # use /etc/pam.d/sshd configuration (the default)
          service_name: "sshd"
+         # use the "auth" modules in the PAM config
+         # "no" by default
```
Comment (webvictim) [inline docs/4.4/features/ssh-pam.md:61]:
```suggestion
         # "false" by default
```

File: docs/4.4/features/ssh-pam.md [hunk 3]
```diff
@@ -286,3 +289,17 @@ ssh_service:
...
+
+## Additional authentication steps
+
+Using the PAM `auth` modules, it is possible to add additional authentication
+steps during user login. These can include passwords, 2nd factor or even
+biometrics.
+
+Note that Teleport enables strong SSH authentication out of the box using
+certificates. For most users, hardening [the initial Teleport
+authentication](../admin-guide#authentication) (e.g. `tsh login`) is preferred.
```
Comment (webvictim) [inline docs/4.4/features/ssh-pam.md:301]:
```suggestion
authentication](../admin-guide.md#authentication) (e.g. `tsh login`) is preferred.
```

**Review by webvictim — COMMENTED** [review]
LGTM

**Comment by webvictim** [comment]
Looks like the test failures are legit - `TestPAM`

**Review by russjones — APPROVED** [review]

**Review by klizhentas — APPROVED** [review]

**Pull request merged**